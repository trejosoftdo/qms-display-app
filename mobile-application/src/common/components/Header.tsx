import * as React from 'react';
import { StyleSheet, View, Image } from 'react-native';
import { Text } from 'react-native-paper';
import icon from '../../../assets/icon.png';
import AppLightTheme from '../theme';

/**
 * HeaderProps defines the props for the Header Component.
 *
 * @property message - the message of the header
 */
interface HeaderProps {
  message: string;
}

/**
 * A component that display a general header for the views in the app
 *
 * @param {HeaderProps} props - The props for the Header component.
 */
const Header: React.FC<HeaderProps> = (props: HeaderProps) => (
  <View style={styles.container}>
    <Image source={icon} style={styles.icon}/>
    <Text style={styles.message} variant="titleMedium">
      {props.message}
    </Text>
  </View>
);

const styles = StyleSheet.create({
  container: {
    backgroundColor: AppLightTheme.colors.background,
    alignItems: 'center',
    justifyContent: 'center',
    width: '100%',
    maxWidth: 500,
  },
  icon: {
    width: 200 ,
    height: 200,
  },
  message: {
    color: AppLightTheme.colors.secondary,
    paddingBottom: 32,
    textAlign: 'justify'
  },
});

export default Header;
