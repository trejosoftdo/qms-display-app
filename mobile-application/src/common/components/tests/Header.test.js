import React from 'react';
import Header from '../Header';

describe('Header component', () => {
  it('renders the component with the provided props as expected', () => {
    expect(renderShallow(<Header message="Test Message" />)).toMatchSnapshot();
  });
});