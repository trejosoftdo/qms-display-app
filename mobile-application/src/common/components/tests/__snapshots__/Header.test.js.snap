// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders the component with the provided props as expected 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "justifyContent": "center",
      "maxWidth": 500,
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      {
        "height": 200,
        "width": 200,
      }
    }
  />
  <ForwardRef(Text)
    style={
      {
        "color": "#3c3744",
        "paddingBottom": 32,
        "textAlign": "justify",
      }
    }
    variant="titleMedium"
  >
    Test Message
  </ForwardRef(Text)>
</View>
`;
